### ISSUE TITLE
Get-AzScheduledQueryRule fails if QueryRule not found

### ISSUE THREAD

**Issue by MrCNeale** [Azure/azure-powershell#20442]
### Description

Running
Get-AzScheduledQueryRule -Name "myRuleName" -ResourceGroupName "myRgName"
Against a specific rule works if the rule exists.
If the rule name is not found then the cmdlet exits incorrectly claiming the cmdlet is not found.
It should exit gracefully without internal exception.

### Issue script & Debug output

```PowerShell
PS /home/cneale3> Get-azscheduledqueryrule -Name "myrule" -ResourceGroupName "cloud-shell-storage-eastus"
DEBUG: 4:10:00 PM - GetAzureRMContextCommand end processing.
DEBUG: [CmdletBeginProcessing]: Starting command
DEBUG: CmdletBeginProcessing: 
DEBUG: CmdletProcessRecordStart: 
DEBUG: CmdletGetPipeline: 
DEBUG: CmdletBeforeAPICall: 
DEBUG: URLCreated: /subscriptions/xxx/resourceGroups/cloud-shell-storage-eastus/providers/Microsoft.Insights/scheduledQueryRules/myrule?api-version=2021-08-01
DEBUG: RequestCreated: /subscriptions/xxx/resourceGroups/cloud-shell-storage-eastus/providers/Microsoft.Insights/scheduledQueryRules/myrule?api-version=2021-08-01
DEBUG: HeaderParametersAdded: 
DEBUG: ============================ HTTP REQUEST ============================

HTTP Method:
GET

Absolute Uri:
https://management.azure.com/subscriptions/xxx/resourceGroups/cloud-shell-storage-eastus/providers/Microsoft.Insights/scheduledQueryRules/myrule?api-version=2021-08-01

Headers:
x-ms-unique-id                : 9
x-ms-client-request-id        : 09e791dc-067c-4f3b-9ed6-dc3654e6b50f
CommandName                   : Get-azscheduledqueryrule
FullCommandName               : Get-AzScheduledQueryRule_Get
ParameterSetName              : __AllParameterSets
User-Agent                    : AzurePowershell/v9.2.0,PSVersion/v7.3.0,Az.ScheduledQueryRule/4.3.0

Body:



DEBUG: BeforeCall: 
DEBUG: ============================ HTTP RESPONSE ============================

Status Code:
NotFound

Headers:
Cache-Control                 : no-cache
Pragma                        : no-cache
x-ms-failure-cause            : gateway
x-ms-request-id               : b88346bb-f6ad-4672-8e30-586696796106
x-ms-correlation-request-id   : b88346bb-f6ad-4672-8e30-586696796106
x-ms-routing-request-id       : WESTEUROPE:20221214T161000Z:b88346bb-f6ad-4672-8e30-586696796106
Strict-Transport-Security     : max-age=31536000; includeSubDomains
X-Content-Type-Options        : nosniff
Date                          : Wed, 14 Dec 2022 16:10:00 GMT

Body:
{
  "error": {
    "code": "ResourceNotFound",
    "message": "The Resource 'Microsoft.Insights/scheduledqueryrules/myrule' under resource group 'cloud-shell-storage-eastus' was not found. For more details please go to https://aka.ms/ARMResourceNotFoundFix"
  }
}


DEBUG: ResponseCreated: 
DEBUG: BeforeResponseDispatch: 
WARNING: Get-AzScheduledQueryRule_Get is not found. The most similar Azure PowerShell command is:
        Get-AzScheduledQueryRule
Get-AzScheduledQueryRule_Get: The Resource 'Microsoft.Insights/scheduledqueryrules/myrule' under resource group 'cloud-shell-storage-eastus' was not found. For more details please go to https://aka.ms/ARMResourceNotFoundFix
DEBUG: [Finally]: Getting exception 'Microsoft.Azure.Commands.Common.Exceptions.AzPSResourceNotFoundCloudException: InternalException' from response
DEBUG: Finally: 
DEBUG: CmdletAfterAPICall: 
DEBUG: [CmdletProcessRecordAsyncEnd]: Finish HTTP process
DEBUG: CmdletProcessRecordAsyncEnd: 
DEBUG: CmdletProcessRecordEnd: 
DEBUG: AzureQoSEvent:  Module: Az.Monitor:4.3.0; CommandName: Get-AzScheduledQueryRule; PSVersion: 7.3.0; IsSuccess: False; Duration: 00:00:00.2212214; Exception: InternalException;
PS /home/cneale3>
```

### Environment data

```PowerShell
$PSVersionTable            

Name                           Value
----                           -----
PSVersion                      7.3.0
PSEdition                      Core
GitCommitId                    7.3.0
OS                             Linux 5.4.0-1098-azure #104~18.04.2-Ubuntu SMP Tue Nov 29 12:13:35 UTC 2022
Platform                       Unix
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

### Module versions

```PowerShell
get-module az*  

ModuleType Version    PreRelease Name                                ExportedCommands
---------- -------    ---------- ----                                ----------------
Script     2.10.4                Az.Accounts                         {Add-AzEnvironment, Clear-AzConfig, Clear-AzContext, Clear-AzDefault…}
Script     5.2.0                 Az.Compute                          {Add-AzImageDataDisk, Add-AzVhd, Add-AzVMAdditionalUnattendContent, Add-AzVMDataDisk…}
Script     4.3.0                 Az.Monitor                          {Add-AzLogProfile, Add-AzMetricAlertRule, Add-AzMetricAlertRuleV2, Add-AzWebtestAlertRule…}
Script     5.2.0                 Az.Network                          {Add-AzApplicationGatewayAuthenticationCertificate, Add-AzApplicationGatewayBackendAddressPool, Add-AzApplicationGatewayBackendHttpSetting, Add-AzApplicationGat…
Script     6.5.0                 Az.Resources                        {Export-AzResourceGroup, Export-AzTemplateSpec, Get-AzDenyAssignment, Get-AzDeployment…}
Script     5.2.0                 Az.Storage                          {Add-AzRmStorageContainerLegalHold, Add-AzStorageAccountManagementPolicyAction, Add-AzStorageAccountNetworkRule, Close-AzStorageFileHandle…}
Script     0.0.0.10              AzureAD.Standard.Preview            {Add-AzureADApplicationOwner, Add-AzureADDeviceRegisteredOwner, Add-AzureADDeviceRegisteredUser, Add-AzureADDirectoryRoleMember…}
Script     0.9.3                 AzurePSDrive
```

### Error output

```PowerShell
Resolve-AzError
DEBUG: 4:12:06 PM - ResolveError begin processing with ParameterSet 'AnyErrorParameterSet'.
DEBUG: 4:12:06 PM - using account id 'MSI@50342'...
DEBUG: 4:12:06 PM - [ConfigManager] Got nothing from [DisplayBreakingChangeWarning], Module = [], Cmdlet = []. Returning default value [True].
WARNING: Upcoming breaking changes in the cmdlet 'Resolve-AzError' :
The `Resolve-Error` alias will be removed in a future release.  Please change any scripts that use this alias to use `Resolve-AzError` instead.
Note : Go to https://aka.ms/azps-changewarnings for steps to suppress this breaking change warning, and other information on breaking changes in Azure PowerShell.

   HistoryId: 25

Message        : [ResourceNotFound] : The Resource 'Microsoft.Insights/scheduledqueryrules/myrule' under resource group 'cloud-shell-storage-eastus' was not found. For more details please go to 
                 https://aka.ms/ARMResourceNotFoundFix
StackTrace     : 
Exception      : System.Exception
InvocationInfo : {Get-AzScheduledQueryRule_Get}
Line           : Get-azscheduledqueryrule -Name "myrule" -ResourceGroupName "cloud-shell-storage-eastus"
Position       : At line:1 char:1
                 + Get-azscheduledqueryrule -Name "myrule" -ResourceGroupName "cloud-she …
                 + ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
HistoryId      : 25


The Azure PowerShell team is listening, please let us know how we are doing: https://aka.ms/azpssurvey?Q_CHL=ERROR.

DEBUG: AzureQoSEvent:  Module: Az.Accounts:2.10.4; CommandName: Resolve-AzError; PSVersion: 7.3.0; IsSuccess: True; Duration: 00:00:00.0235828
DEBUG: 4:12:06 PM - [ConfigManager] Got [True] from [EnableDataCollection], Module = [], Cmdlet = [].
DEBUG: 4:12:06 PM - ResolveError end processing.
```

**Comment by dingmeng-xue** [comment]
Thanks for reporting. We agree with your suggestion. It is a bug in our code. We will fix it.

**Comment by isra-fel** [comment]
Fixed by #19228 
The updated version will be available early March.
